app/controllers/api/v1/cancellation_reasons_controller.rb

```diff
@@ -21,6 +16,13 @@ def show
       def serializer
         Api::V1::CancellationReasonSerializer
       end
+
+      def cancellation_reasons
+        @cancellation_reasons.cancellation_reasons_by({
+          ids: params["ids"],
+          offer: params["offer"]
```

Comment:
Rather than `offer` this should be more generic. At the moment we have `offer` and `order` but imagine if we want to add a third or forth option. Could change it to something more like `for` where the allowed options are only `offer` and `order` - sanitize the params

Reply:
Once you've made the above change, I would strongly advocate to remove the `cancellation_reasons` method and put the code into the `index` method to make the code more readable.